packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -0,0 +1,210 @@
+import { 
+  RDSClient,
+  CreateDBInstanceCommand,
+  waitUntilDBInstanceAvailable,
+  DeleteDBInstanceCommand,
+  waitUntilDBInstanceDeleted 
+} from "@aws-sdk/client-rds";
+import { EC2Client, AuthorizeSecurityGroupIngressCommand, RevokeSecurityGroupIngressCommand } from '@aws-sdk/client-ec2';
+import { knex } from 'knex';
+
+const DEFAULT_DB_INSTANCE_TYPE = "db.m5.large";
+const DEFAULT_DB_STORAGE = 8;
+const DEFAULT_SECURITY_GROUP = "default";
+
+export const createRDSInstance = async (config: {
+  identifier: string,
+  engine: 'mysql',
+  dbname: string,
+  username: string,
+  password: string,
+  region: string,
+  instanceClass?: string,
+  storage?: number,
+}) => {
+  const client = new RDSClient({ region: config.region });
+  const params = {
+    /** input parameters */
+    "DBInstanceClass": config.instanceClass ?? DEFAULT_DB_INSTANCE_TYPE,
+    "DBInstanceIdentifier": config.identifier,
+    "AllocatedStorage": config.storage ?? DEFAULT_DB_STORAGE,
+    "Engine": config.engine,
+    "DBName": config.dbname,
+    "MasterUsername": config.username,
+    "MasterUserPassword": config.password,
+  };
+  const command = new CreateDBInstanceCommand(params);
+  
+  try {
+    await client.send(command);
+    const availableResponse = await waitUntilDBInstanceAvailable(
+      {
+        maxWaitTime: 3600,
+        maxDelay: 120,
+        minDelay: 60,
+        client,
+      },
+      {
+        DBInstanceIdentifier: config.identifier,
+      },
+    );
+
+    if (availableResponse.state !== 'SUCCESS') {
+      throw new Error("Error in creating a new RDS instance.");
+    }
+
+    const dbInstance = availableResponse.reason.DBInstances[0];
+    if (!dbInstance) {
+      throw new Error("RDS Instance details are missing.");
+    }
+
+    return {
+      endpoint: dbInstance.Endpoint.Address,
+      port: dbInstance.Endpoint.Port,
+      dbName: dbInstance.DBName,
+    };
+  } catch (error) {
+    console.log(error);
+    throw new Error("Error in creating RDS instance.");
+  }
+};
+
+export const deleteDBInstance = async (identifier: string, region: string) => {
+  const client = new RDSClient({ region });
+  const params = {
+    "DBInstanceIdentifier": identifier,
+    "SkipFinalSnapshot": true,
+  };
+  const command = new DeleteDBInstanceCommand(params);
+  try {
+    await client.send(command);
+    
+    // TODO: Revisit the below logic for waitUntilDBInstanceDeleted.
```

Comment:
When are we revisiting this? What's blocking us getting this in place today?

Reply:
I see some issues with the way SDK method 'waitUntilDBInstanceDeleted' works. It polls for the instance status at regular intervals and returns once the status is 'DELETED'. But the problem is once we trigger the instance deletion, it takes few minutes for the instance to be actually deleted. SDK returns 'DELETED' status only for few seconds after instance deletion. After that, it returns 'NO SUCH INSTANCE' error. I can reduce the interval to a lower value and get it to work most of the times but still there are changes of flaky test.

With current implementation, we trigger the deletion and exit the test. If test doesn't fail at this step, the instance will be deleted few minutes (usually 2-3 minutes) after the test run.